syft/frameworks/torch/mpc/fss.py

```diff
@@ -0,0 +1,401 @@
+"""
+This is an implementation of Function Secret Sharing
+
+Useful papers are:
+- Function Secret Sharing- Improvements and Extensions, Boyle 2017 https://eprint.iacr.org/2018/707.pdf
+- Secure Computation with Preprocessing via Function Secret Sharing, Boyle 2019 https://eprint.iacr.org/2019/1095
+
+Note that the protocols are quite different in aspect from those papers
+"""
+import hashlib
+
+import torch as th
+import syft as sy
+from syft.execution.plan import func2plan
+from syft.generic.frameworks.hook.trace import tracer
+from syft.workers.base import BaseWorker
+
+
+λ = 110  # 6  # 110 or 63  # security parameter
+n = 32  # 8  # 32  # bit precision
+dtype = th.int32
+
+no_wrap = {"no_wrap": True}
+
+
+def initialize_crypto_plans(worker):
+    """
+    This is called manually for the moment, to build the plan used to perform
+    Function Secret Sharing on a specific worker.
+    """
+    eq_plan_1 = sy.Plan(
+        forward_func=lambda x, y: mask_builder(x, y, "eq"),
+        owner=worker,
+        tags=["#fss_eq_plan_1"],
+        is_built=True,
+    )
+    worker.register_obj(eq_plan_1)
+    eq_plan_2 = sy.Plan(
+        forward_func=eq_eval_plan, owner=worker, tags=["#fss_eq_plan_2"], is_built=True
+    )
+    worker.register_obj(eq_plan_2)
+
+    comp_plan_1 = sy.Plan(
+        forward_func=lambda x, y: mask_builder(x, y, "comp"),
+        owner=worker,
+        tags=["#fss_comp_plan_1"],
+        is_built=True,
+    )
+    worker.register_obj(comp_plan_1)
+    comp_plan_2 = sy.Plan(
+        forward_func=comp_eval_plan, owner=worker, tags=["#fss_comp_plan_2"], is_built=True
+    )
+    worker.register_obj(comp_plan_2)
+
+    xor_add_plan = sy.Plan(
+        forward_func=xor_add_convert_1, owner=worker, tags=["#xor_add_1"], is_built=True
+    )
+    worker.register_obj(xor_add_plan)
+    xor_add_plan = sy.Plan(
+        forward_func=xor_add_convert_2, owner=worker, tags=["#xor_add_2"], is_built=True
+    )
+    worker.register_obj(xor_add_plan)
+
+
+def request_run_plan(worker, plan_tag, location, return_value, args=tuple(), kwargs=dict()):
```

Comment:
Random thought here.
Isn't it possible to use pointer to plans to request a run?

Reply:
Yes you can, but you need an extra round of communication to get your pointer